app/views.py

```diff
@@ -86,10 +86,68 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class EditProfileView(LoginRequiredMixin, UpdateView):
+class SocialLinkFormMixin(FormMixin):
+    def get_class_name(self):
+        return (
+            self.model if self.model else self.object.__class__ if self.object else self.classname
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context["has_social_links_form"] = True
+
+        initial_data = {
+            "object_id": 0 if not self.object else self.object.pk,
```

Comment:
Regarding the earlier comment, `0` is used as a placeholder if the `object_id` is unknown as `None` throws a required field error (thread of questioning is also in the same earlier comment).